app/services/ItemsCachingService.scala

```diff
@@ -0,0 +1,179 @@
+                  ),
+                  Classification(item.cusCode, identificationTypeCode = Some(CUSCode))
+                ) ++
+                  item.nationalAdditionalCodes.map(
+                    code => Classification(Some(code), identificationTypeCode = Some(NationalAdditionalCode))
+                  ) ++ item.taricAdditionalCodes
+                  .map(code => Classification(Some(code), identificationTypeCode = Some(TARICAdditionalCode)))
+              )
+            )
+        )
+      )
+
+  private def createGoodsItem(seq: Int, cachedData: CacheMap): GovernmentAgencyGoodsItem = {
+    val itemTypeData = goodsItemFromItemTypes(cachedData)
+    val commodity = itemTypeData.fold(Commodity())(_.commodity.getOrElse(Commodity()))
+    val updatedCommodity = commodity.copy(goodsMeasure = commodityGoodsMeasure(cachedData).flatMap(_.goodsMeasure))
+
+    GovernmentAgencyGoodsItem(
+      sequenceNumeric = seq + 1,
+      statisticalValueAmount = itemTypeData.flatMap(_.statisticalValueAmount),
+      packagings = generatePackages(cachedData).getOrElse(Seq.empty),
+      governmentProcedures = procedureCodes(cachedData).getOrElse(Seq.empty),
+      commodity = Some(updatedCommodity),
+      additionalInformations = additionalInfo(cachedData).getOrElse(Seq.empty),
+      additionalDocuments = documents(cachedData).getOrElse(Seq.empty)
+    )
+  }
+  /*
+    Fetch cache elements of all the pages using goodsItemCacheId for pages procedure codes, itemType, packaging, goods measure,
+    additional information , documents produced.
+    create related wco-dec elements
+    create goods item for the cache and append to items already added in the cache
+    save updated items to cache and remove the elements in goodsItemCacheId
+   */
+  def addItemToCache(
+    goodsItemCacheId: String,
+    supplementaryCacheId: String
+  )(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Boolean] =
+    cacheService.fetch(goodsItemCacheId).flatMap {
+      case Some(cachedData) =>
+        cacheService
+          .fetchAndGetEntry[Seq[GovernmentAgencyGoodsItem]](supplementaryCacheId, itemsId)
+          .flatMap(
+            items =>
+              cacheService
+                .cache[Seq[GovernmentAgencyGoodsItem]](
+                  supplementaryCacheId,
+                  itemsId,
+                  items.getOrElse(Seq.empty) :+ createGoodsItem(items.fold(0)(_.size), cachedData)
+                )
+                .flatMap(_ => cacheService.remove(goodsItemCacheId).map(_.status == 204))
+          )
+      case None => Future.successful(false)
+    }
+
+}
+
+object ExportsItemsCacheIds {
+  val itemsCachePages = Map(
+    PackageInformation.formId -> "package",
+    ProcedureCodesData.formId -> "procedure codes",
+    CommodityMeasure.commodityFormId -> "goods measure",
+    DocumentsProducedData.formId -> "additional documents",
+    ItemType.id -> "Item type",
+    AdditionalInformationData.formId -> "additional information"
```

Comment:
First, here is a good place to move ID from `CacheIdGenerator`.
Second, please use one approach start with capital letter or lower case